src/fastoad/io/formatter.py

```diff
@@ -25,7 +30,7 @@ class IVariableIOFormatter(ABC):
     """
 
     @abstractmethod
-    def read_variables(self, data_source: Union[str, IO]) -> VariableList:
+    def read_variables(self, data_source: Union[str, PathLike, IO, IOBase]) -> VariableList:
```

Comment:
Here, why are both `IO` and `IOBase` needed ? Also, why isn't `TextIOBase` used instead of `IO` ?